File: app/src/androidTest/java/me/varlez/movies/MockTestRunner.java
package me.varlez.movies;

import android.app.Application;
import android.content.Context;
import android.support.test.runner.AndroidJUnitRunner;

/**
 * Custom test runner that will be using our mocked Application.
 */
public class MockTestRunner extends AndroidJUnitRunner {
    @Override
    public Application newApplication(ClassLoader cl, String className, Context context) throws InstantiationException, IllegalAccessException, ClassNotFoundException {
        return super.newApplication(cl, MockMoviesApp.class.getName(), context);
    }
}

File: app/src/androidTest/java/me/varlez/movies/MockTestRunner.kt
package me.varlez.movies

import android.app.Application
import android.content.Context
import android.support.test.runner.AndroidJUnitRunner

/**
 * Custom test runner that will be using our mocked Application.
 */
class MockTestRunner : AndroidJUnitRunner() {
    @Throws(InstantiationException::class, IllegalAccessException::class, ClassNotFoundException::class)
    override fun newApplication(cl: ClassLoader, className: String, context: Context): Application {
        return super.newApplication(cl, MockMoviesApp::class.java.name, context)
    }
}
